content/guides/destructuring.adoc

```diff
@@ -211,9 +259,38 @@ When you have nested vectors, you can use `:as` or `&` at any level as well.
 ;= 10 20 [10 20]
 ----
 
+or even:
+
+[source,clojure]
+----
+(def my-multidimensional-line [[1 2 5 0 8] [6 4 3 1 9]])
+(let [[[x0 & coords0 :as orig] [x1 & coords1 :as dest] :as line ] my-multidimensional-line]
+  (println line)
+  (println orig)
+  (println dest)
+  (println x0 coords0)
+  (println x1 coords1))
+;= [[1 2 5 0 8] [6 4 3 1 9]]
+;= [1 2 5 0 8]
+;= [6 4 3 1 9]
+;= 1 (2 5 0 8)
+;= 6 (4 3 1 9)
+----
+
+In particular, one can further destructure whatever value has been bound using `&`:
+
+[source,clojure]
+----
+(let [[[x0 & [y0 & coords0]] [x1 & [y1 & coords1]]] my-multidimensional-line]
+  (println "(" x0 "," y0 ") followed by" coords0)
+  (println "(" x1 "," y1 ") followed by" coords1))
+( 1 , 2 ) followed by (5 0 8)
+( 6 , 4 ) followed by (3 1 9)
+----
+
 == Associative Destructuring
 
-Associative destructuring is similar to sequential destructuring, but applied instead to associative (key-value) structures (including maps, records, vectors, etc). The associative bindings are concerned with concisely extracting values of the map by key.
+Associative destructuring is similar to sequential destructuring, but applied instead to associative (key-value) structures (maps and records). The associative bindings are concerned with concisely extracting values of the map by key.
```

Comment:
Why the change here? Vectors are associative (index to element) and work fine with associative destructuring.

Reply:
That didn't occur to me. I think we need an example.